### PR TITLE
Implement `myqueries` page using routify

### PR DESCRIPTION
**Pull request by dkumiszhan**
Client side rendering is used for rendering the page.
There are some TODOs related to CORS and making Golang backend
and Svelte frontend work together. I will tackle those later
in separate PRs.

This PR contains the first commit from https://github.com/honeynet/ochi/pull/76. I will rebase once the other PR is merged.

### CODE REVIEW

**Comment by glaslos** [comment]
Conflicts need to be resolved.

**Comment by dkumiszhan** [comment]
> Conflicts need to be resolved.

done

File: backend/handlers.go [hunk 1]
```diff
@@ -124,3 +125,103 @@ func (cs *server) loginHandler(w http.ResponseWriter, r *http.Request, _ httprou
...
+	if err = json.NewEncoder(w).Encode(queries); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+// createQueryHandler creates a new query.
+func (cs *server) createQueryHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	userID := r.Context().Value(userID("userID")).(string)
+	decoder := json.NewDecoder(r.Body)
```
Comment (glaslos) [inline backend/handlers.go:150]: If you read from `r.Body` you should also do a `defer r.Body.Close()`

File: backend/handlers.go [hunk 2]
```diff
@@ -124,3 +125,103 @@ func (cs *server) loginHandler(w http.ResponseWriter, r *http.Request, _ httprou
...
+	q, err := cs.queryRepo.GetByID(id)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	if userID != q.OwnerID {
+		http.Error(w, "Unauthorized", http.StatusInternalServerError)
+		return
+	}
+	decoder := json.NewDecoder(r.Body)
```
Comment (glaslos) [inline backend/handlers.go:183]: close `r.Body`

File: backend/handlers.go [hunk 3]
```diff
@@ -124,3 +125,103 @@ func (cs *server) loginHandler(w http.ResponseWriter, r *http.Request, _ httprou
...
+	if id != t.ID {
+		http.Error(w, "Ids dont match", http.StatusBadRequest)
+		return
+	}
+	err = cs.queryRepo.Update(t.ID, t.Content, t.Description, t.Active)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
```
Comment (glaslos) [inline backend/handlers.go:199]: consider returning the updated query so the client can display it

File: backend/handlers.go [hunk 4]
```diff
@@ -124,3 +125,103 @@ func (cs *server) loginHandler(w http.ResponseWriter, r *http.Request, _ httprou
...
+func (cs *server) deleteQueryHandler(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+	userID := r.Context().Value(userID("userID")).(string)
+	id := p.ByName("id")
+	q, err := cs.queryRepo.GetByID(id)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	if userID != q.OwnerID {
+
```
Comment (glaslos) [inline backend/handlers.go:212]: too much space :)

File: package-lock.json
Comment (glaslos) [inline package-lock.json:1]: is this expected to have that many changes or is this due to testing different routing libraries?

File: tsconfig.json [hunk 5]
```diff
@@ -9,5 +9,6 @@
     "public/*"
   ],
   "compilerOptions": {
+    "allowJs": true
```
Comment (glaslos) [inline tsconfig.json:12]: what do you need this for?

**Review by glaslos — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 5 above)
Comment (dkumiszhan) [reply tsconfig.json:12]: This was needed to import routify generated file which is in javascript.

File: backend/handlers.go (hunk 2 above)
Comment (dkumiszhan) [reply backend/handlers.go:184]: done

File: backend/handlers.go (hunk 1 above)
Comment (dkumiszhan) [reply backend/handlers.go:150]: done

File: backend/handlers.go (hunk 3 above)
Comment (dkumiszhan) [reply backend/handlers.go:201]: Let me do it later

File: backend/handlers.go (hunk 4 above)
Comment (dkumiszhan) [reply backend/handlers.go]: done

File: package-lock.json
Comment (dkumiszhan) [reply package-lock.json:1]: Yes, this is expected. I did reinstall and confirmed.

**Review by glaslos — APPROVED** [review]
LGTM. Sorry for the late review!

**Pull request merged**